libparsec/crates/client/src/workspace/transactions/move_entry.rs

```diff
@@ -242,11 +242,13 @@ async fn move_entry_same_parent(
 ) -> Result<(), WorkspaceMoveEntryError> {
     let parent_id = parent_manifest.base.id;
     let mut_parent = Arc::make_mut(&mut parent_manifest);
+    let mut data = HashMap::with_capacity(2);
 
-    let child_id = match mut_parent.children.remove(&src_child_name) {
-        None => return Err(WorkspaceMoveEntryError::SourceNotFound),
-        Some(child_id) => child_id,
+    let Some(child_id) = mut_parent.children.get(&src_child_name).copied() else {
+        return Err(WorkspaceMoveEntryError::SourceNotFound);
     };
+    // Remove the source child from the parent's `children` field
+    data.insert(src_child_name.clone(), None);
```

Comment:
Maybe add this as well, since you seem to do it in all non-error cases:
```
data.insert(dst_child_name, Some(child_id));
```

Reply:
Done